src/Driver/Mysqli/Exception/NonStreamResourceUsedAsLargeObject.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Driver\Mysqli\Exception;
+
+use Doctrine\DBAL\Driver\AbstractException;
+
+use function sprintf;
+
+/**
+ * @internal
+ *
+ * @psalm-immutable
+ */
+final class NonStreamResourceUsedAsLargeObject extends AbstractException
+{
+    public static function new(int $parameter): self
+    {
+        return new self(sprintf('A non-stream resource passed as a LARGE_OBJECT parameter #%d', $parameter));
```

Comment:
This sentence feels a bit weird to me, I'd either use
```suggestion
        return new self(sprintf('A non-stream resource was passed as a LARGE_OBJECT parameter #%d', $parameter));
```

or 

```suggestion
        return new self(sprintf('Non-stream resource passed as a LARGE_OBJECT parameter #%d', $parameter));
```

Reply:
What if we get rid of the past tense here and change it to something like: 'The resource passed as a LARGE_OBJECT parameter must be of type "stream"?' It will be more about how to fix the problem, not just the problem statement.

Reply:
As a side note, have you seen any articles/guides specifically on composing error messages? E.g. in navigation it seems acceptable to sacrifice the grammatical correctness of sentence in favor of brevity: do not lean on door). And many other computer programs I saw (especially in the past) would use this principle as well.

Reply:
As another side note, we probably can get rid of this exception and let the underlying driver decide whether it can bind this resource to the statement or not. Then the message (if it's clear enough) may come directly from the driver (should be easier to google).

Reply:
> As a side note, have you seen any articles/guides specifically on composing error messages? E.g. in navigation it seems acceptable to sacrifice the grammatical correctness of sentence in favor of brevity: do not lean on door). And many other computer programs I saw (especially in the past) would use this principle as well.

I didn't but it's an interesting topic for sure!

Reply:
Just [googled](https://forum.wordreference.com/threads/do-not-lean-on-door.2337460/) it for myself and will leave it here :

> Signs often have reduced grammar, as do newspaper headlines, captions of photographs, and so on. They may omit articles and auxiliary verbs, for example. We're so used to this that it might actually sound strange to use a complete sentence in such situations.

This is by no means a justification for my illiteracy.

Reply:
Haha I stumbled upon this blog post that refers to us while looking this up: http://rosstuck.com/formatting-exception-messages

This one is quite interesting: https://softwareengineering.stackexchange.com/questions/29433/how-to-write-a-good-exception-message, but more about the contents than the form.

This one advocates against brevity: http://amortizedcost.net/writing-good-exception-message/

IMO there might have been a point in brevity in the past, when memory was an issue, but nowadays we might want to revisit that thought and start using multiline exception messages, which detail the context, the problem and solutions as suggested in the second link.